src/translator/text-based-field.ts

```diff
@@ -1,12 +1,35 @@
-import { EElementType, IValidation, TElement } from "src/context-providers";
-import { createConditionalRenderingObject } from "./helper";
+import {
+    EElementType,
+    IColumns,
+    IValidation,
+    TElement,
+} from "src/context-providers";
+import { ELEMENT_VALIDATION_TYPES } from "src/data";
+import { SimpleIdGenerator } from "src/util/simple-id-generator";
+import {
+    ISchemaCondition,
+    ISchemaConditionalRendering,
+    createConditionalRenderingObject,
+    translateConditionalRenderingObject,
+} from "./helper";
 
 export namespace TextBasedField {
     export interface ISchemaValidation {
         [key: string]: string | boolean;
         errorMessage?: string;
     }
 
+    export interface IElementSchema {
+        columns: IColumns;
+        label: string;
+        placeholder: string;
+        required: boolean;
+        requiredErrorMessage: string;
+        uiType: EElementType;
+        validation?: ISchemaValidation[];
+        showIf?: ISchemaCondition[];
+    }
```

Comment:
this interface is not necessary as we should use FEE schema where possible, can replace with `ITextFieldSchema | IEmailFieldSchema | INumericFieldSchema`